webapp/store/views.py

```diff
@@ -318,6 +318,15 @@ def details_library(entity_name, library_name):
 
     library = publisher_api.get_charm_library(entity_name, library_id)
 
+    function_args = {}
+    for text in re.split("\n+", library["content"]):
+        stripped_text = text.strip()
+        if stripped_text.startswith("def"):
+            stripped__arr = stripped_text.split("(")
+            func_name = stripped__arr[0].split(" ")[1]
+            func_args = stripped__arr[1].split(")")[0]
+            function_args[func_name] = func_args
+
```

Comment:
Is the parser not handling that for you?

Reply:
unfortunately not

Reply:
I think this logic should be added in here https://github.com/canonical-web-and-design/canonicalwebteam.docstring-extractor/blob/main/canonicalwebteam/docstring_extractor/__init__.py#L24

This will make this code available in case we start using this parser in other projects.

Reply:
@jkfran but do I have access in the `docstring_extractor` to the file content?

Reply:
It needs to be done in the `docstring_extractor`. Yes, you have access to the code, instead of doing it like this, using `ast` will be the neater way.